pkg/apiserver/rest/asset_scan_controller.go

```diff
@@ -45,6 +45,14 @@ func (s *ServerImpl) PostAssetScans(ctx echo.Context) error {
 		return sendError(ctx, http.StatusBadRequest, fmt.Sprintf("failed to bind request: %v", err))
 	}
 
+	status, ok := assetScan.GetResourceCleanupStatus()
```

Comment:
```suggestion

	status, ok := assetScan.GetResourceCleanupStatus()
	switch {
	case !ok:
		return sendError(ctx, http.StatusBadRequest, "invalid request: resource cleanup status is missing")
	case status.State == models.ResourceCleanupStatusStatePending:
		fallthrough
	case status.State == models.ResourceCleanupStatusStateSkipped:
		break
	default:
		return sendError(ctx, http.StatusBadRequest, fmt.Sprintf("invalid request: initial state for resource cleanup status is invalid: %s", status.State))
	}
```

Alternative approach for this switch-case which allows to handle the `Pending` and `Skipped` cases differently. (Which is not necessary for now.)